### PR TITLE
Log a warning when a play has isApplicable() as true and invariantHolds() as false

### PR DESCRIPTION
**Pull request by JoshWong16**
### Description

Log a warning when play::isApplicable() is true and play::invariantHolds() is false. Meant to detect unwanted behaviour.

### Testing Done

Manual testing using AI vs AI and forcing invariantHolds() to be false. The new warning appeared when it was supposed to. 

### Resolved Issues

https://github.com/UBC-Thunderbots/Software/issues/2248

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by akhilveeraghanta — COMMENTED** [review]
LGTM :100:! Looks like you just need to run `fix_formatting.sh` in the `environment_setup` folder

File: src/software/ai/hl/stp/stp.cpp [hunk 1]
```diff
@@ -123,6 +123,10 @@ std::unique_ptr<Play> STP::calculateNewPlay(const World& world)
         auto play = play_constructor(play_config);
         if (play->isApplicable(world))
         {
+            if (!play->invariantHolds(world)){
+                LOG(WARNING) << "Play::isApplicable() is true and Play::invariantHolds() is false"
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/stp.cpp]: please also print the name of the play with `objectTypeName(*play)`

**Review by jonathanlew — CHANGES_REQUESTED** [review]

**Comment by jonathanlew** [comment]
Please also change the name of the PR to be more descriptive

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**